src/core/unit.c

```diff
@@ -3616,6 +3616,8 @@ void unit_catchup(Unit *u) {
 
         if (UNIT_VTABLE(u)->catchup)
                 UNIT_VTABLE(u)->catchup(u);
+
+        unit_cgroup_catchup(u);
```

Comment:
I suggest adding a guard
```
if (UNIT_HAS_CGROUP_CONTEXT(u))
```
here (or inside `unit_cgroup_catchup`). a) Makes no sense on cgroup-less units, b) save some work during reload/reexec.

Reply:
ack, i added it into `unit_cgroup_catchup`, and also added a check in `unit_check_cgroup_events` to avoid unneeded work if `!u->cgroup_path`, like is already done for the other `unit_check_*` functions. thanks!